ingestion/src/metadata/great_expectations/column_test_builders.py

```diff
@@ -0,0 +1,188 @@
+            ometa_conn: OMeta API connection
+            table_entity: table entity for the test
+        """
+        self.result = result
+        self.ometa_conn = ometa_conn
+        self.table_entity = table_entity
+        self.timestamp = Timestamp(__root__=int(int(datetime.now().timestamp())))
+
+    def add_test(self) -> None:
+        """Send an API request to add a test result to a table entity"""
+        self.ometa_conn.add_column_test(self.table_entity, self._build_test())
+
+    @staticmethod
+    def build_test_case(config, test_type) -> ColumnTestCase:
+        """Build test case based on the test type
+
+        Args:
+            config: any instance of a column test case
+            test_type: any instance of a column test type
+        Return:
+            ColumnTestCase
+        """
+        return ColumnTestCase(
+            config=config,
+            columnTestType=test_type,
+        )
+
+    def build_test_case_results(self) -> TestCaseResult:
+        """Build test case result base on GE test result"""
+        return TestCaseResult(
+            executionTime=self.timestamp,
+            testCaseStatus=TestCaseStatus.Success
+            if self.result["success"]
+            else TestCaseStatus.Failed,
+            result="Failing rows percentage: "
+            f"{self.result['result']['unexpected_percent']}",
+        )
+
+    def build_test_request(self, *, config, test_type) -> CreateColumnTestRequest:
+        """Build a test case request to add the test to the tabe
+
+        Args:
+            test_case: test case
+            test_case_result: a test case result
+        Return:
+            CreateColumnTestRequest
+        """
+        return CreateColumnTestRequest(
+            columnName=self.result["expectation_config"]["kwargs"]["column"],
+            testCase=self.build_test_case(config=config, test_type=test_type),
+            result=self.build_test_case_results(),
+            updatedAt=self.timestamp,
+        )
+
+    @abstractmethod
+    def _build_test(self) -> CreateColumnTestRequest:
+        """Used to create the column test request for the specific test.
+        Needs to be implemented by the specific builder.
+        """
+
+
+class ColumnValuesLengthsToBeBetweenBuilder(BaseColumnTestBuilder):
+    """Builder for `expect_column_value_lengths_to_be_between` GE expectation"""
+
+    def _build_test(self) -> CreateColumnTestRequest:
+        """Specific test builder for the test"""
+        return self.build_test_request(
+            config=columnValuesLengthsToBeBetween.ColumnValueLengthsToBeBetween(
+                minLength=self.result["expectation_config"]["kwargs"].get("min_value"),
+                maxLength=self.result["expectation_config"]["kwargs"].get("max_value"),
+            ),
+            test_type=ColumnTestType.columnValueLengthsToBeBetween,
+        )
+
+
+class ColumnValuesToBeBetweenBuilder(BaseColumnTestBuilder):
```

Comment:
Hi @TeddyCr, overall this looks great. Amazing job, big thanks 🙏 

One question/suggestion I'd have here would be about the structure. As the number of tests to support will increase greatly, would it make sense to have each Builder in its own module? IDK how big this file can get and might be hard to control what we have in scope

Reply:
Yes that makes a lot of sense. 🙂